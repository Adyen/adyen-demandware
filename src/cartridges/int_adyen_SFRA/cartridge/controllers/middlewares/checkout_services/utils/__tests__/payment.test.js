const handlePaymentAuthorization = require('../payment');

const adyenHelpers = require('*/cartridge/scripts/checkout/adyenHelpers');

let res;
let emit;
beforeEach(() => {
  jest.clearAllMocks();
  res = { json: jest.fn() };
  emit = jest.fn();
});

describe('Payment', () => {
  it('should return json with error details when payment is unsuccessful', () => {
    adyenHelpers.handlePayments.mockReturnValue({ error: true });
    const paymentCompleted = handlePaymentAuthorization({}, { res }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should return 3ds2 json response', () => {
    adyenHelpers.handlePayments.mockReturnValue({
      threeDS2: true,
      resultCode: 'IdentifyShopper',
      action: 'mocked_action',
    });
    const paymentCompleted = handlePaymentAuthorization({}, { res }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should return 3d json response', () => {
    adyenHelpers.handlePayments.mockReturnValue({
      redirectObject: {
        url: 'mocked_url',
        data: { PaReq: 'mocked_PaReq', MD: 'mocked_MD' },
      },
      signature: 'mocked_signature',
      authorized3d: true,
    });
    const paymentCompleted = handlePaymentAuthorization({}, { res }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should redirect', () => {
    adyenHelpers.handlePayments.mockReturnValue({
      redirectObject: { url: 'mocked_url' },
      authorized3d: false,
      signature: 'mocked_signature',
    });
    const paymentCompleted = handlePaymentAuthorization({}, { res }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should not redirect', () => {
    adyenHelpers.handlePayments.mockReturnValue({
      redirectObject: false,
      threeDS2: false,
    });
    const paymentCompleted = handlePaymentAuthorization({}, { res }, emit);
    expect(paymentCompleted).toBeTruthy();
  });
});
