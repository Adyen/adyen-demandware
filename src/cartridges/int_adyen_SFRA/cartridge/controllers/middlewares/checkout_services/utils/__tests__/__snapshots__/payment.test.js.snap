// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should redirect 1`] = `
Array [
  Array [
    Object {
      "continueUrl": "[\\"Adyen-Redirect\\",\\"redirectUrl\\",\\"mocked_url\\",\\"signature\\",\\"mocked_signature\\"]",
      "error": false,
    },
  ],
]
`;

exports[`Payment should return 3d json response 1`] = `
Array [
  Array [
    Object {
      "continueUrl": "[\\"Adyen-Adyen3D\\",\\"IssuerURL\\",\\"mocked_url\\",\\"PaRequest\\",\\"mocked_PaReq\\",\\"MD\\",\\"mocked_MD\\"]",
      "error": false,
    },
  ],
]
`;

exports[`Payment should return 3ds2 json response 1`] = `
Array [
  Array [
    Object {
      "continueUrl": "[\\"Adyen-Adyen3DS2\\",\\"resultCode\\",\\"IdentifyShopper\\",\\"token3ds2\\",\\"mocked_token\\"]",
      "error": false,
    },
  ],
]
`;

exports[`Payment should return json with error details when payment is unsuccessful 1`] = `
Array [
  Array [
    Object {
      "error": true,
      "errorMessage": "mocked_error.payment.not.valid",
    },
  ],
]
`;
